java/ql/test/library-tests/frameworks/android/taint-database/Sinks.java

```diff
@@ -213,147 +213,147 @@ public static void query(MyContentResolver target) {
 		SQLiteDatabase db = taint();
 		ContentValues values = taint();
-		String selection = taint();
-		String[] selectionArgs = taint();
+		String selection = taint(); // $hasTaintFlowSink
+		String[] selectionArgs = {taint()};
 		target.update(db, values, selection, selectionArgs);
 	}
 
 	public static void update(SQLiteDatabase target) {
-		String table = taint();
+		String table = taint(); // $hasTaintFlowSink
 		ContentValues values = taint();
-		String whereClause = taint();
-		String[] whereArgs = taint();
+		String whereClause = taint(); // $hasTaintFlowSink
+		String[] whereArgs = {taint()};
 		target.update(table, values, whereClause, whereArgs);
 	}
 
 	public static void update(MyContentResolver target) {
 		Uri uri = taint();
 		ContentValues values = taint();
-		String selection = taint();
-		String[] selectionArgs = taint();
+		String selection = taint(); // $hasTaintFlowSink
+		String[] selectionArgs = {taint()};
 		target.update(uri, values, selection, selectionArgs);
 	}
 
 	public static void update(MyContentProvider target) {
 		Uri uri = taint();
 		ContentValues values = taint();
-		String selection = taint();
-		String[] selectionArgs = taint();
+		String selection = taint(); // $hasTaintFlowSink
+		String[] selectionArgs = {taint()};
 		target.update(uri, values, selection, selectionArgs);
 	}
 
 	public static void updateWithOnConflict(SQLiteDatabase target) {
-		String table = taint();
+		String table = taint(); // $hasTaintFlowSink
 		ContentValues values = taint();
-		String whereClause = taint();
-		String[] whereArgs = taint();
+		String whereClause = taint(); // $hasTaintFlowSink
+		String[] whereArgs = {taint()};
 		int conflictAlgorithm = taint();
 		target.updateWithOnConflict(table, values, whereClause, whereArgs, conflictAlgorithm);
 	}
 
 	public static void queryNumEntries() {
 		SQLiteDatabase db = taint();
-		String table = taint();
-		String selection = taint();
+		String table = taint(); // $hasTaintFlowSink
+		String selection = taint(); // $hasTaintFlowSink
 		DatabaseUtils.queryNumEntries(db, table, selection);
 	}
 
 	public static void queryNumEntries2() {
 		SQLiteDatabase db = taint();
-		String table = taint();
-		String selection = taint();
-		String[] selectionArgs = taint();
+		String table = taint(); // $hasTaintFlowSink
+		String selection = taint(); // $hasTaintFlowSink
+		String[] selectionArgs = {taint()};
 		DatabaseUtils.queryNumEntries(db, table, selection, selectionArgs);
 	}
 
 	public static void createDbFromSqlStatements() {
 		Context context = taint();
 		String dbName = taint();
 		int dbVersion = taint();
-		String sqlStatements = taint();
+		String sqlStatements = taint(); // $hasTaintFlowSink
 		DatabaseUtils.createDbFromSqlStatements(context, dbName, dbVersion, sqlStatements);
 	}
 
 	public static void blobFileDescriptorForQuery() {
 		SQLiteDatabase db = taint();
-		String query = taint();
-		String[] selectionArgs = taint();
+		String query = taint(); // $hasTaintFlowSink
```

Comment:
Similarly here, I think the label could be a bit clearer. E.g. "taintReachesQueryInjectionSink".